### DISCUSSION TITLE
Retries config

### DISCUSSION THREAD

**Discussion by javmezac** [bee-queue/bee-queue#679] [Q&A]
I am setting the retry config to 3 attempts but it seems that the failed job is being retried 4 times, this is causing that when I do this:

```
queue.on('failed', (job, err) => {
       console.log(job.options.retries);
   });
```

the following is printed:

2
1,
0,
0

this is how i am setting the retries option:

```
const addToQueue = async (queueName, data, retries = 3) => {
  const queue = getQueue(queueName);
  return queue
    .createJob({ name: queueName, ...data })
    .retries(retries)
    .save();
};
```

Any help with this please?

**Comment by demipixel** [comment]
Retries is how many times it tries again after the first attempt. The default is `0` which means it doesn't retry and only make the first attempt.

If the first attempt fails, and all three retries fail, that would emit 4 failures total.

As per [the docs](https://github.com/bee-queue/bee-queue#retrying), it seems like `retries` is decremented before `retrying` and `failed` are called. Not sure if this is the best behavior because, as you can see, there doesn't seem to be a way to differentiate between the penultimate and last attempt, but that's how it currently works.
